salt/runners/spacewalk.py

```diff
@@ -180,7 +180,14 @@ def api(server, command, *args, **kwargs):
         log.error(err_msg)
         return {call: err_msg}
 
-    namespace, method = command.split(".")
+    command = command.split(".")
+    if len(command) == 1:
+        return {
+            call: "Error: command must use the following format: 'namespace.method'"
+        }
+
+    method = command.pop()
+    namespace = ".".join(command)
```

Comment:
This could be simplified somewhat:

```suggestion
    namespace, _, command = command.rpartition(".")
    if namespace:
        return {
            call: "Error: command must use the following format: 'namespace.method'"
        }
```

Reply:
that is elegant! Did not know about `rpartition`, thanks :+1:

Reply:
FWIW most string methods have equivalent `r` methods that do things from the right side: split, rsplit, partition, rpartition, find, rfind, etc.

Granted, split and rsplit are identical if you don't provide a `maxsplit`, but... yeah, lots of super useful string methods :slightly_smiling_face: